src/binaryutils.cpp

```diff
@@ -232,24 +234,35 @@ static Status validateTensor(const std::shared_ptr<TensorInfo>& tensorInfo,
         return StatusCode::INVALID_SHAPE;
     }
 
-    if (checkBatchSizeMismatch(tensorInfo, src.contents().bytes_contents_size())) {
+    size_t content_size = (buffer == nullptr) ? src.contents().bytes_contents_size() : 1;
+    if (checkBatchSizeMismatch(tensorInfo, content_size)) {
         SPDLOG_DEBUG("Input: {} request batch size is incorrect. Expected: {} Actual: {}",
             tensorInfo->getMappedName(),
             tensorInfo->getBatchSize().has_value() ? tensorInfo->getBatchSize().value().toString() : std::string{"none"},
             src.contents().bytes_contents_size());
         return StatusCode::INVALID_BATCH_SIZE;
     }
 
-    for (int i = 0; i < src.contents().bytes_contents_size(); i++) {
-        if (src.contents().bytes_contents(i).size() <= 0) {
-            return StatusCode::BYTES_CONTENTS_EMPTY;
-        }
+    if (content_size <= 0) {
```

Comment:
+spdlog with explanation

Reply:
check if checkBatchSizeMismatch will always fail when input batchsize is zero. Tensor info can have 0 as a dynamic bs